app/models/rental.rb

```diff
@@ -40,12 +46,24 @@ def renter_is_assignable
   scope :inactive_rentals, -> { where(rental_status: %w(canceled dropped_off)) }
   scope :rented_by, ->(user) { where(renter_id: user) }
   scope :created_by, ->(user) { where(creator_id: user) }
+
+  #   if these queries ever become a problem this would be faster, i think a cross apply or something like that would probably be even faster.
+  #   select * from rentals where
+  #       ((select sum(amount) from financial_transactions where rental_id=rentals.id and
+  #         (transactable_type='Payment' or transactable_type='Cancelation'))
+  #       -(select sum(amount) from financial_transactions where rental_id=rentals.id and
+  #         not (transactable_type='Payment' or transactable_type='Cancelation')))
+  #       > 0
   scope :with_balance_due, -> { Rental.where id: Rental.select { |rental| rental.balance.positive? }.collect(&:id) }
-  scope :with_balance_over, ->(min) { Rental.where id: Rental.select { |rental| rental.balance >= min }.collect(&:id) }
+  scope :with_balance_over, ->(min) { Rental.where id: Rental.select { |rental| rental.balance > min }.collect(&:id) }
```

Comment:
So, can I ask, as a design decision, why this PR's code passes around objects' IDs after every query instead of just the objects themselves?

Reply:
In this case I think its because I wanted an active record relation not an array.  This could easily be rewritten it is already inefficient as it is.